### ISSUE TITLE
Create unit tests for new fields

### ISSUE THREAD

**Issue by icebob** [icebob/vue-form-generator#40]
- [x] fieldCleave
- [x] fieldMasked
- [x] fieldNoUiSlider
- [x] fieldPikaday
- [x] fieldSlider
- [x] fieldSpektrum
- [x] fieldVueMultiSelect
- [x] fieldGoogleAddress

**Comment by lionel-bijaoui** [comment]
I'm going to try after I read a bit more about Karma

**Comment by icebob** [comment]
OK, but firstly I have to solve the external lib loading problem.

**Comment by icebob** [comment]
I added external libs to karma. So now can be improve new field unit tests. https://github.com/icebob/vue-form-generator/commit/776f6c479fec8105df55d69288a310cf58b2386c

**Comment by lionel-bijaoui** [comment]
(I was about to suggest that :smile: )

---
> Passing arrow functions (“lambdas”) to Mocha is discouraged

From the [doc](http://mochajs.org/)
Should we change that ?

**Comment by icebob** [comment]
Oooo.... what?!?!

**Comment by lionel-bijaoui** [comment]
basically, don't do this:
```
describe('my suite', () => {
  it('my test', () => {
    // should set the timeout of this test to 1000 ms; instead will fail
    this.timeout(1000);
    assert.ok(true);
  });
});
```
but this:
```
describe('my suite', function() {
  it('my test', function() {
    this.timeout(1000);
    assert.ok(true);
  });
});
```

**Comment by icebob** [comment]
You can use arrows because we use webpack as preprocessor and webpack use babel

**Comment by lionel-bijaoui** [comment]
Unit test are going through Webpack ? Wow ok, I didn't know... Webpack does everything :astonished:

**Comment by icebob** [comment]
yes :) https://github.com/icebob/vue-form-generator/blob/master/test/unit/karma.conf.js#L35

**Comment by lionel-bijaoui** [comment]
Sorry, I missed that totally

**Comment by icebob** [comment]
Nop, so you can use any ES6 feature

**Comment by lionel-bijaoui** [comment]
Sweet!

**Comment by icebob** [comment]
what field test are you make? Because I plan to make an other field tests

**Comment by lionel-bijaoui** [comment]
I'm starting with #36

**Comment by icebob** [comment]
**What we need to test in this external fields:**

- [ ] check HTML elements (example: https://github.com/icebob/vue-form-generator/blob/master/test/unit/specs/fields/fieldSelect.spec.js#L48 )

- [ ] after init check the lib read the value from model and value is correct (example: https://github.com/icebob/vue-form-generator/blob/master/test/unit/specs/fields/fieldSelect.spec.js#L63 )

- [ ] if model value change, lib got the new value (example: https://github.com/icebob/vue-form-generator/blob/master/test/unit/specs/fields/fieldSelect.spec.js#L78 )

- [ ] if value changed (via HTML element, maye need a trigger with js), check the model got the value (example: https://github.com/icebob/vue-form-generator/blob/master/test/unit/specs/fields/fieldSelect.spec.js#L87 )

**Comment by icebob** [comment]
I make `fieldMasked` and `fieldSpektrum`

**Comment by lionel-bijaoui** [comment]
I'm going to try `fieldVueMultiSelect` even if `noUiSlider` is not finished

**Comment by icebob** [comment]
Ok, for multiselect I recommend you to use `fieldSelect.spec.js` as a base

**Comment by lionel-bijaoui** [comment]
Roger that ! :)

**Comment by lionel-bijaoui** [comment]
#46 I did something wrong with the branch (a bad rebase), so this one contain a lot a commit from #44.
I hope it's ok.

**Comment by icebob** [comment]
Ok, #46 & #44 merged

**Comment by lionel-bijaoui** [comment]
It was too early for #44, I made several mistakes that I'm fixing

**Comment by icebob** [comment]
Ok, just &46 contains #44 because both was in the same branch, so I can only merge both

**Comment by lionel-bijaoui** [comment]
My bad sorry about that

**Comment by icebob** [comment]
Nop :)

**Comment by icebob** [comment]
fieldMasked done.

**Comment by icebob** [comment]
fieldPikaday, fieldSlider done.

**Comment by icebob** [comment]
fieldSpectrum done.

**Comment by icebob** [comment]
fieldGoogleAddress "done"

**Comment by lionel-bijaoui** [comment]
Wow cool, you are on :fire:

**Comment by icebob** [comment]
Now I start the last fieldCleave too.

**Comment by icebob** [comment]
All test done. :tada: 

I made some changes, so if you open the karma debug page ( http://localhost:9877/debug.html ) for test debugging, you will see a similar page instead of empty white blank page :smile:  
![image](https://cloud.githubusercontent.com/assets/306521/18007842/f41e1174-6ba5-11e6-8511-a0190494555c.png)

**Comment by lionel-bijaoui** [comment]
That's so cool ! I need your help with the doc: 
- [ ] visible & disabled, values schema functions description
- [ ] fieldSubmit
- [ ] customizable styles

I hope we can finish everything early next week.

**Comment by icebob** [comment]
Merged #51
